### PR TITLE
Document proxy configuration for OAuth 2.0 client

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: docs/src/main/sphinx/security/oauth2.rst [hunk 1]
```diff
@@ -108,6 +108,12 @@ The following configuration properties are available:
        for more information.
    * - ``http-server.authentication.oauth2.principal-field``
      - The field of the access token used for the Trino user principal. Defaults to ``sub``. Other commonly used fields include ``sAMAccountName``, ``name``, ``upn``, and ``email``.
+   * - ``oauth2-jwk.http-client.http-proxy``
+     - Proxy host and port in host:port format. Use this property to connect
+       to an OAuth2 endpoint via a HTTP proxy.
+   * - ``oauth2-jwk.http-client.http-proxy.secure``
+     - Set this to true if we connecting to to an OAuth2 endpoint via a
```
Comment (mosabua) [inline docs/src/main/sphinx/security/oauth2.rst:115]: Identify if HTTP proxy for OAuth2 endpoint access is secured. Defaults to `false`.

Comment (mosabua) [reply docs/src/main/sphinx/security/oauth2.rst:115]: Related question though .. what do you mean by secured HTTP proxy?

File: docs/src/main/sphinx/security/oauth2.rst [hunk 2]
```diff
@@ -108,6 +108,12 @@ The following configuration properties are available:
        for more information.
    * - ``http-server.authentication.oauth2.principal-field``
      - The field of the access token used for the Trino user principal. Defaults to ``sub``. Other commonly used fields include ``sAMAccountName``, ``name``, ``upn``, and ``email``.
+   * - ``oauth2-jwk.http-client.http-proxy``
```
Comment (lukasz-walkiewicz) [inline docs/src/main/sphinx/security/oauth2.rst:111]: Why only these two? Both of them come from airlift http client configuration which provides many more useful options (https://github.com/airlift/airlift/blob/master/http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java) which probably should also be documented. Trino uses quite a few instances of http client, each with it's own prefix. How about documenting all of http client configuration properties in one place and just referencing it here?

Comment (Praveen2112) [reply docs/src/main/sphinx/security/oauth2.rst:111]: I was thinking about the same ? Should it be a separate documentation for HTTP client configuration ?

File: docs/src/main/sphinx/security/oauth2.rst (hunk 1 above)
Comment (Praveen2112) [reply docs/src/main/sphinx/security/oauth2.rst:115]: HTTPS proxy.

Comment (mosabua) [reply docs/src/main/sphinx/security/oauth2.rst:115]: Please update as suggested

**Comment by bitsondatadev** [comment]
:wave: @Praveen2112 - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**